src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfo.cs

```diff
@@ -443,8 +438,8 @@ internal string GetDebugInfo(int indent = 0)
         }
 #endif
 
-        internal bool HasGetter { get; set; }
-        internal bool HasSetter { get; set; }
+        internal bool HasGetter => _untypedGet is not null;
```

Comment:
I guess me must have simplified something which makes it not needed anymore. I initially implemented HasGetter similar way several weeks back and was getting some failures because of that

Reply:
PropertyInfoForTypeInfo is a bit of a beast to get rid of, we will need quite a bit of refactoring